lib/internal/process.js

```diff
@@ -145,7 +145,69 @@ function setupKillAndExit() {
       process._exiting = true;
       process.emit('exit', process.exitCode || 0);
     }
+
+    // Flush stdio streams prior to exit.
+    // `flushSync` not present if stream redirected to file in shell.
+    flushSync(process.stdout);
+    flushSync(process.stderr);
+
     process.reallyExit(process.exitCode || 0);
+
+    function flushSync(stream) {
+
+      // Behavior of this function outside of process.exit() is undefined
+      // due to the following factors:
+      // * Stream fd may be blocking after this call.
+      // * In the event of an incomplete flush pending buffered write
+      //   requests may be truncated.
+      // * No return code.
+
+      if (stream._writev)
+        return;
```

Comment:
> In my defence this issue involved quite a few subsystems, so please forgive me for not understanding how it all works.

@kzc I think node is too big for anyone to understand how it all works on all its details.

I think `_writev` should be handled, __but__ in the worst case we should put a big comment in how we skip streams with `_writev`.